src/main/java/me/desair/tus/server/TusFileUploadService.java

```diff
@@ -98,8 +99,11 @@ public TusFileUploadService withUploadLockingService(final UploadLockingService
 
     public TusFileUploadService withStoragePath(final String storagePath) {
         Validate.notBlank(storagePath, "The storage path cannot be blank");
-        withUploadStorageService(new DiskStorageService(idFactory, storagePath));
-        withUploadLockingService(new DiskLockingService(idFactory, storagePath));
+        CachedDiskAndLockingStorageService cachedDiskAndLockingStorageService =
```

Comment:
It would be nice if we could make caching optional (not default) using a new builder method `withThreadLocalCache()`. That method would wrap the current `uploadStorageService` and `uploadLockingService` values in a new `CachedDiskAndLockingStorageService` instance.

Reply:
Make sure to also enable caching and/or add caching specific tests to the `ITTusFileUploadService`.